### PR TITLE
OptiType

### PR DESCRIPTION
**Pull request by jj-umn**
OptiType: precision HLA typing from next-generation sequencing data

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/optitype/optitype.xml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+[ilp]
+
+# A Pyomo-supported ILP solver. The solver must be globally accessible in the
+# environment OptiType is run, so make sure to include it in PATH.
+# Note: this is NOT a path to the solver binary, but a keyword argument for
+# Pyomo. Examples: glpk, cplex, cbc.
+
+solver=$solver
+threads=1
```
Comment (bgruening) [inline tools/optitype/optitype.xml:59]: should this always stay at 1?

File: tools/optitype/optitype.xml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        </param>
+        <param name="use_discordant" type="boolean" truevalue="true" falsevalue="false" checked="false" label="use_discordant">
+            <help><![CDATA[
+We call a read pair discordant if its two ends best-map to two disjoint sets
+of alleles. Such reads can be either omitted or either of their ends treated
+as unpaired hits. Note: discordant read pairs are reported on the coverage
+plots as unpaired reads, regardless of this setting. ]]>
+            </help>
+        </param>
+        <param name="outdir" type="hidden" value="output_dir"/>
```
Comment (bgruening) [inline tools/optitype/optitype.xml:140]: why is this needed as param? You can just set this in the commandline.

File: tools/optitype/optitype.xml [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+            <conditional name="fastq_input">
+                <param name="fastq_input_selector" value="paired"/>
+                <param name="fastq_input1" ftype="fastqsanger" value="exome/NA11995_SRR766010_1_fished.fastq"/>
+                <param name="fastq_input2" ftype="fastqsanger" value="exome/NA11995_SRR766010_2_fished.fastq"/>
+            </conditional>
+            <param name="read_type" value="--dna"/>
+            <param name="unpaired_weight" value="0.2"/>
+            <output name="result">
+                <assert_contents>
+                    <has_text text="A*01:01" />
```
Comment (bgruening) [inline tools/optitype/optitype.xml:170]: can we also check for the number of columns and maybe rows? This is also possible using the asserts.

File: tools/optitype/optitype.xml [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+**OptiType**
+============
+
+OptiType_ is a novel HLA genotyping algorithm based on integer linear programming, capable of producing accurate 4-digit HLA genotyping predictions from NGS data by simultaneously selecting all major and minor HLA Class I alleles.
+
+**INPUTS** 
+
+     RNA or DNA sequences in fastq format.
+
+**OPTIONS**
```
Comment (bgruening) [inline tools/optitype/optitype.xml:186]: I would remove those options here, its written above under each parameter where it belongs.

**Review by bgruening — COMMENTED** [review]
Cool, thanks @jj-umn! Small review inline.

File: tools/optitype/optitype.xml (hunk 4 above)
Comment (jj-umn) [reply tools/optitype/optitype.xml:186]: Will be removed.

File: tools/optitype/optitype.xml (hunk 2 above)
Comment (jj-umn) [reply tools/optitype/optitype.xml:140]: will be removed.

File: tools/optitype/optitype.xml (hunk 1 above)
Comment (jj-umn) [reply tools/optitype/optitype.xml:59]: Yes. the threads=1 has to be included and all listed solvers are single threaded.  
Only the glpk solver is included in conda optitype=1.3.5

**Comment by bgruening** [comment]
Thanks @jj-umn!

**Review by bgruening — APPROVED** [review]
